aspnetcore/security/anti-request-forgery.md

```diff
@@ -1,18 +1,311 @@
+
+* GET
+* HEAD
+* OPTIONS
+* TRACE
+
+The ASP.NET Core team recommends you use ``AutoValidateAntiforgeryToken`` broadly for non-API scenarios, so that your POST actions are protected by default (APIs don't have an automatic mechanism for sending the non-cookie part of the token). The alternative is to ignore antiforgery tokens by default, unless ``ValidateAntiForgeryToken`` is applied to the individual action method. It's more likely in this scenario for a POST action method to be left unprotected, leaving your app vulnerable to CSRF attacks. Even anonymous POSTS should send the antiforgery token.
+
+Example (class level):
+<!-- literal_block {"xml:space": "preserve", "language": "c#", "dupnames": [], "linenos": false, "classes": [], "ids": [], "backrefs": [], "highlight_args": {}, "names": []} -->
+
+```c#
+[Authorize]
+[AutoValidateAntiforgeryToken]
+public class ManageController : Controller
+{
+```
+
+Example (globally):
+<!-- literal_block {"xml:space": "preserve", "language": "c#", "dupnames": [], "linenos": false, "classes": [], "ids": [], "backrefs": [], "highlight_args": {}, "names": []} -->
+
+```c#
+services.AddMvc(options => 
+    options.Filters.Add(new AutoValidateAntiforgeryTokenAttribute()));
+```
+
+### IgnoreAntiforgeryToken
+
+The ``IgnoreAntiforgeryToken`` filter is used to eliminate the need for an antiforgery token to be present for a given action (or controller). When applied, this filter will override ``ValidateAntiForgeryToken`` and/or ``AutoValidateAntiforgeryToken`` filters specified at a higher level (globally or on a controller).
+
+<!-- literal_block {"xml:space": "preserve", "language": "c#", "dupnames": [], "linenos": false, "classes": [], "ids": [], "backrefs": [], "highlight_args": {}, "names": []} -->
+
+```c#
+[Authorize]
+[AutoValidateAntiforgeryToken]
+public class ManageController : Controller
+{
+  [HttpPost]
+  [IgnoreAntiforgeryToken]
+  public async Task<IActionResult> DoSomethingSafe(SomeViewModel model)
+  {
+    // no antiforgery token required
+  }
+}
+```
+
+## JavaScript, AJAX, and SPAs
+
+In traditional HTML-based applications, antiforgery tokens are passed to the server using hidden form fields. In modern JavaScript-based apps and single page applications (SPAs), in which AJAX requests are sent programmatically from scripts, other techniques may be used to send the token. If cookies are used to store authentication tokens and to authenticate API requests on the server, then CSRF will be a potential problmem. However, if local storage is used to store the token, CSRF vulnerability may be mitigated, since values from local storage are not sent automatically to the server with every new request.
```

Comment:
>In modern JavaScript-based apps and single page applications (SPAs), in which AJAX requests are sent programmatically from scripts, other techniques may be used to send the token. 

too long
How about:
In apps using AJAX requests , other techniques  must be used. For example in SPAs and ...